Analytics-CSharp/Segment/Analytics/Plugins/UserInfoPlugin.cs

```diff
@@ -0,0 +1,48 @@
+using Segment.Serialization;
+using Segment.Sovran;
+
+namespace Segment.Analytics.Plugins
+{
+    /// <summary>
+    /// Analytics plugin used to populate events with basic UserInfo data.
+    /// Auto-added to analytics client on construction
+    /// </summary>
+    public class UserInfoPlugin : Plugin
+    {
+        public override PluginType type => PluginType.Before;
+
+        private Store store;
+
+        public override void Configure(Analytics analytics)
+        {
+            base.Configure(analytics);
+            store = analytics.store;
+        }
+
+        private void ApplyUserInfoData(RawEvent @event)
+        {
+            if (@event is IdentifyEvent identifyEvent)
+            {
+                analytics.userInfo.userId = identifyEvent.userId;
+                analytics.userInfo.anonymousId = identifyEvent.anonymousId;
```

Comment:
we don't change anouymousId in IdentifyEvent. also we need to do null checks for userid and traits before assigning, since identify takes userid or traits or both